### DISCUSSION TITLE
dt in Value Tracker fails

### DISCUSSION THREAD

**Discussion by Pasquale19** [ManimCommunity/manim#3993] [Q&A] [answered]
Hey I tried to set a ValueTracker with am updater that is using the ` dt` parameter. However it seems like the Value Tracker is only updated once and not after each frame. I also tried to use a DecimalNumber and a Variable without any difference. After different attempts my current workaround is to use the `self.renderer.time` variable. I thought the `dt` parameter should be updatet after each frame or did I misunderstood something?

Here is a short example that show that the Value Tracker isn't updated by using the dt parameter, but using the self.renderer.time works.

```
from manim import *

class Updater_dt_Scene(Scene):
    def construct(self):
        #self.always_update_mobjects=True
        #dt 
        value_tracker1 = ValueTracker(0)
        value_tracker1.add_updater(lambda m,dt: m.set_value(dt))
        self.add(value_tracker1)
        dec_number_tracker1=DecimalNumber(value_tracker1.get_value(),num_decimal_places=2,include_sign=True)
        dec_number_tracker1.add_updater(lambda m: m.set_value(value_tracker1.get_value()))
        dec_number1=DecimalNumber(value_tracker1.get_value(),num_decimal_places=2,include_sign=True)
        dec_number1.add_updater(lambda mob,dt: mob.set_value(dt))
        var1=Variable(0,Text("var"),num_decimal_places=2)
        var1.add_updater(lambda mob,dt: mob.tracker.set_value(dt))
        vg1=VGroup(Text("dt"),dec_number_tracker1,dec_number1,var1)
        vg1.arrange(DOWN).shift(LEFT*3)
        self.add(vg1)

        #self.renderer.time
        value_tracker2 = ValueTracker(0)
        value_tracker2.add_updater(lambda m: m.set_value(self.renderer.time))
        self.add(value_tracker2)
        dec_number_tracker2=DecimalNumber(value_tracker2.get_value(),num_decimal_places=2,include_sign=True)
        dec_number_tracker2.add_updater(lambda m: m.set_value(value_tracker2.get_value()))
        dec_number2=DecimalNumber(0,num_decimal_places=2,include_sign=True)
        dec_number2.add_updater(lambda mob: mob.set_value(self.renderer.time))
        var2=Variable(0,Text("var"),num_decimal_places=2)
        var2.add_updater(lambda mob: mob.tracker.set_value(self.renderer.time))
        vg2=VGroup(Text("self.renderer.time"),dec_number_tracker2,dec_number2,var2)
        vg2.arrange(DOWN).shift(RIGHT*3)
        vg2.stroke_color=BLUE
        self.add(vg2)
        self.wait(3,frozen_frame=False)
```

https://github.com/user-attachments/assets/beee04d5-05ab-4f44-a58c-dda05534a9be

**Comment by uwezi** [comment, marked as answer]
`dt` is the time difference between the current and the previous frame. It is equal to 1/framerate.

With `value_tracker1.add_updater(lambda m,dt: m.set_value(dt))` you are setting the value of the value tracker to the value `dt` over and over again. Try `value_tracker1.add_updater(lambda m,dt: m.increment_value(dt))` for a cumulative updating...

Over on Discord I could have directly demonstrated this in a live code example... 
https://docs.manim.community/en/stable/faq/general.html?highlight=discord#where-can-i-find-more-resources-for-learning-manim

**Reply by Pasquale19** [reply]
Thank you very much for this answer. Now I understand it.  I was wondering about this for such a long time.  I thougt it was the time since the animation started.
